services/galley/src/Galley/API/Internal.hs

```diff
@@ -438,29 +441,53 @@ sitemap = do
 
 rmUser :: UserId -> Maybe ConnId -> Galley ()
 rmUser user conn = do
-  let n = unsafeRange 100 :: Range 1 100 Int32
-  tids <- Data.teamIdsForPagination user Nothing (rcast n)
+  let n = toRange (Proxy @100) :: Range 1 100 Int32
+      nRange1000 = rcast n :: Range 1 1000 Int32
+  tids <- Data.teamIdsForPagination user Nothing n
   leaveTeams tids
-  cids <- Data.conversationIdRowsForPagination user Nothing (rcast n)
-  leaveConversations user cids
+  localDomain <- viewFederationDomain
+  allConvIds <- Query.conversationIdsPageFrom user (GetPaginatedConversationIds Nothing nRange1000)
+  goConvPages (toLocal (Qualified user localDomain)) nRange1000 allConvIds
   Data.eraseClients user
   where
+    goConvPages :: Local UserId -> Range 1 1000 Int32 -> ConvIdsPage -> Galley ()
+    goConvPages lusr range page = do
+      localDomain <- viewFederationDomain
+      let (remoteConvs, localConvs) = partitionRemoteOrLocalIds' localDomain . mtpResults $ page
+      leaveLocalConversations localConvs
+      leaveRemoteConversations lusr remoteConvs
+      when (mtpHasMore page) $ do
+        nextState <- case (mtpsState . mtpPagingState) page of
+          Nothing -> pure Nothing
+          Just bs -> case parseConversationPagingState bs of
+            Left _ -> throwErrorDescriptionType @BadPagingState
+            Right p -> pure . Just $ p
+        let usr = lUnqualified lusr
+            nextQuery = GetPaginatedConversationIds nextState range
+        newCids <- Query.conversationIdsPageFrom usr nextQuery
```

Comment:
@pcapriotti or @jschaul , can you take a look at this? I changed this after getting a PR approval, because the underlying paging things changed on `develop` in the meantime. I am not sure if I'm doing the paging state threading properly, let alone if there's a nicer way to do it.